### PR TITLE
[SwapModal]: Fix for can approve Swap with no amount entered Fix/15553

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
#15553 , #15605 
### What does the PR do

reset output params when any of the values in the form changes and 
wrong value when amount entered with more that 4 decimal places in SwapSignModal

### Affected areas

SwapModal.qml, SwapSignModal.qml

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/user-attachments/assets/cb27ce1f-126f-4daa-b450-d29a8d492738

https://github.com/user-attachments/assets/2c8868cf-7cb5-4d17-9933-64318e2adfa6

### CODE REVIEW

**Review by alexjba — APPROVED** [review]
LGTM! Tested the input 👍

Can't test the signing on testnet

File: ui/app/AppLayouts/Wallet/popups/swap/SwapSignModal.qml [hunk 1]
```diff
@@ -45,8 +47,8 @@ SignTransactionModalBase {
 
     title: qsTr("Sign Swap")
     //: e.g. (swap) 100 DAI to 100 USDT
-    subtitle: qsTr("%1 %2 to %3 %4").arg(formatBigNumber(fromTokenAmount, 4)).arg(fromTokenSymbol) // FIXME get the correct number of decimals to display from the "symbol"
-        .arg(formatBigNumber(toTokenAmount, 4)).arg(toTokenSymbol)
+    subtitle: qsTr("%1 to %2").arg(root.currencyStore.formatCurrencyAmount(fromTokenAmount, fromTokenSymbol)) // FIXME get the correct number of decimals to display from the "symbol"
```
Comment (dlipicar) [inline ui/app/AppLayouts/Wallet/popups/swap/SwapSignModal.qml:50]: the comment is no longer valid, right? we're using the correct format with this function

**Review by dlipicar — APPROVED** [review]

**Comment by virginiabalducci** [comment]
Hi @Khushboo-dev-cpp 

The specific issues fixed on this PR looks good. 

I'm seeing that the error of "insufficient funds to pay for gas" is not triggering on this build. Should this functionality work under the scope of this build? 

Thanks!

https://github.com/user-attachments/assets/8832cac7-0aeb-47ca-bd38-1e7adebb5aa9


```
ERR 2024-07-19 11:31:04.878-03:00 rpc response error                         topics="rpc" tid=2097501 file=core.nim:36 err="\nstatus-go error [methodName:wallet_createMultiTransaction, code:-32000, message:eth-archival.rpc.grove.city.error: Insufficient funds for gas * price + value, mainnet.infura.io.error: insufficient funds for gas * price + value: balance 900000000000000, tx cost 5564942461207360, overshot 4664942461207360 ]\n"
ERR 2024-07-19 11:31:04.878-03:00 error doing rpc request                    topics="rpc" tid=2097501 file=core.nim:39 methodName=wallet_createMultiTransaction exception="\nstatus-go error [methodName:wallet_createMultiTransaction, code:-32000, message:eth-archival.rpc.grove.city.error: Insufficient funds for gas * price + value, mainnet.infura.io.error: insufficient funds for gas * price + value: balance 900000000000000, tx cost 5564942461207360, overshot 4664942461207360 ]\n"
```

**Comment by Khushboo-dev-cpp** [comment]
> Hi @Khushboo-dev-cpp
> 
> The specific issues fixed on this PR looks good.
> 
> I'm seeing that the error of "insufficient funds to pay for gas" is not triggering on this build. Should this functionality work under the scope of this build?
> 
> Thanks!
> 
>  Screen.Recording.2024-07-19.at.11.30.29.AM.mov 
> ```
> ERR 2024-07-19 11:31:04.878-03:00 rpc response error                         topics="rpc" tid=2097501 file=core.nim:36 err="\nstatus-go error [methodName:wallet_createMultiTransaction, code:-32000, message:eth-archival.rpc.grove.city.error: Insufficient funds for gas * price + value, mainnet.infura.io.error: insufficient funds for gas * price + value: balance 900000000000000, tx cost 5564942461207360, overshot 4664942461207360 ]\n"
> ERR 2024-07-19 11:31:04.878-03:00 error doing rpc request                    topics="rpc" tid=2097501 file=core.nim:39 methodName=wallet_createMultiTransaction exception="\nstatus-go error [methodName:wallet_createMultiTransaction, code:-32000, message:eth-archival.rpc.grove.city.error: Insufficient funds for gas * price + value, mainnet.infura.io.error: insufficient funds for gas * price + value: balance 900000000000000, tx cost 5564942461207360, overshot 4664942461207360 ]\n"
> ```

Hey hey thats a scenario that shouldn't occur ideally! we should not allow user to sap if he doesnt have sufficient gas or balance so I think this is a bug, this PR doesn't cover it at all. But I think we should create a separate PR and look at it asap.
FYI @dlipicar

**Comment by dlipicar** [comment]
> > Hi @Khushboo-dev-cpp
> > The specific issues fixed on this PR looks good.
> > I'm seeing that the error of "insufficient funds to pay for gas" is not triggering on this build. Should this functionality work under the scope of this build?
> > Thanks!
> > Screen.Recording.2024-07-19.at.11.30.29.AM.mov
> > ```
> > ERR 2024-07-19 11:31:04.878-03:00 rpc response error                         topics="rpc" tid=2097501 file=core.nim:36 err="\nstatus-go error [methodName:wallet_createMultiTransaction, code:-32000, message:eth-archival.rpc.grove.city.error: Insufficient funds for gas * price + value, mainnet.infura.io.error: insufficient funds for gas * price + value: balance 900000000000000, tx cost 5564942461207360, overshot 4664942461207360 ]\n"
> > ERR 2024-07-19 11:31:04.878-03:00 error doing rpc request                    topics="rpc" tid=2097501 file=core.nim:39 methodName=wallet_createMultiTransaction exception="\nstatus-go error [methodName:wallet_createMultiTransaction, code:-32000, message:eth-archival.rpc.grove.city.error: Insufficient funds for gas * price + value, mainnet.infura.io.error: insufficient funds for gas * price + value: balance 900000000000000, tx cost 5564942461207360, overshot 4664942461207360 ]\n"
> > ```
> 
> Hey hey thats a scenario that shouldn't occur ideally! we should not allow user to sap if he doesnt have sufficient gas or balance so I think this is a bug, this PR doesn't cover it at all. But I think we should create a separate PR and look at it asap. @dlipicar will #15668 cover it? maybe yes cause we'll get proposal with this error and disable swap button accordingly?

@virginiabalducci yeap, we're handling all error types other than "entered amount > balance" in my PR

**Review by virginiabalducci — APPROVED** [review]
QA passed

**Pull request merged**